### ISSUE TITLE
Checkbox for Woocommerce not working 

### ISSUE THREAD

**Issue by faraday815** [dannyvankooten/mailchimp-for-wordpress#67]
Hi,

I have the latest version of Mailchimp for WordPress.

I have created a form and set it up on my website http://freshfoodbox.in/get-updates/ - that works perfectly.

However, the checkbox for woocommerce doesn't seem to be working.

I can see a checkbox at the correct position on the checkout page. It is checked. When I submit an order, there are no new subscribers in my mailchimp account (even a day after testing this). I tried both double-opt in and single opt-in.

Any help is very appreciated.

https://wordpress.org/plugins/mailchimp-for-wp/

PS I submitted this on the plugin's wordpress support page, then noticed that you asked for bugs to be posted here.

**Comment by dannyvankooten** [comment]
Hi @faraday815 ,

Did you select a MailChimp list in **MailChimp for WP > Checkboxes** and does this list require any other fields besides an EMAIL or a NAME field?

**Comment by faraday815** [comment]
Hi Danny,

Thanks for getting back to me. I am really struggling to solve this, and am
manually adding customers for the moment.

Yes, I have selected a mailchimp list.
Here are the settings I have on the *Checkboxes* page:

Mailchimp Lists: Selected
Double Opt In: No
Add checkbox to these forms: Selected all of them, including Woocommerce
checkout
Checkbox label text: Send me emails with updates and promotions *(this
shows up fine on the checkout page)*
Pre-check the checkbox: Yes
Load some default CSS: Yes
Checkbox position: After billing details


The list has four fields:

Email (this is the only required field)
First Name
Last Name
Phone Number

The tags are EMAIL, FNAME, LNAME, MMERGE3

Perhaps I must combine First & Last Name into one field called NAME?


I'd love to solve this.

Thanks,
Arjun

**Comment by dannyvankooten** [comment]
Hi Arjun,

That all sounds about right. The plugin automatically fills the **FNAME** and **LNAME** field based on the customer data so that should not be the issue.

My colleague @iMazed will run some extensive tests using your configuration later this week, it's running a little late now. She'll keep you updated! :) Thank you for the detailed report so far.

**Comment by iMazed** [comment]
Yes, I'm on it! First thing that comes to mind for me is that Mailchimp is rejecting the request for some reason. So far I haven't been able to reproduce your issue, but I'm now getting really curious as to what's causing this. Would you mind contacting me through support@mc4wp.com? I'd like to ask a thing or two privately.

**Comment by faraday815** [comment]
Hi Danny,

As a note, the fix that Ines suggested to me yesterday didn't work for me.

**Comment by dannyvankooten** [comment]
Hi @faraday815 ,

Could you give the development version of the plugin a try? You can download it by using the following link: https://github.com/dannyvankooten/mailchimp-for-wordpress/archive/master.zip .

You can then update your plugin to the development version of MailChimp for WordPress by using the [Easy Theme & Plugin Updates](https://wordpress.org/plugins/easy-theme-and-plugin-upgrades/) plugin.

If that version still doesn't work for you, which payment method are you using when the sign-up fails?

**Comment by faraday815** [comment]
Hi @dannyvankooten @iMazed ,

Just got around to this.

I noticed you had an update on 13 Feb which supposedly solved this. I updated, made a test transaction (payment option: Cash on Delivery), but it didn't add the email to my mailchimp list.

I then deleted the plugin, installed the dev version that you sent me, and placed another order (Payment Option: Cheque). This time, it worked.

Another test. With the dev version (which I suppose is exactly the same as the latest version) I placed another order using Cash On Delivery as the payment option. This time again, the mail got added to the list (the first time it worked with Cash on Delivery as an option).

Final test. This time with Direct Bank Transfer as an option. It worked.

Any idea why it might have not worked with the updated plugin version?
